### PR TITLE
move invalid emails out of firestore

### PR DESCRIPTION
**Pull request by goldblatt**
moves the invalid emails as we run into them and places them into a different location. We could just delete them, but idk i got anxious.

### CODE REVIEW

File: scripts/alert_emails/send_emails.ts [hunk 1]
```diff
@@ -196,24 +200,33 @@ async function setLastSnapshotNumber(
...
-            .catch(err => {
-              console.log(err);
-              errorCount += 1;
+            .catch(async (err: CampaignMonitorError) => {
+              // if the email is invalid, move the document to the invalid collection
+              if (err.Code === CM_INVALID_EMAIL_ERROR_CODE && err.Message === CM_INVALID_EMAIL_MESSAGE) {
+                invalidEmailCount += 1;
+                const currentData = (await db.collection("alerts-subscriptions").doc(doc.id).get()).data();
+                if (currentData) {
+                  await db.collection("invalid-alert-subscriptions").doc(doc.id).set(currentData)
```
Comment (goldblatt) [inline scripts/alert_emails/send_emails.ts:209]: changethis to invalid-alerts-subscriptions

File: scripts/alert_emails/send_emails.ts [hunk 2]
```diff
@@ -186,7 +190,7 @@ async function setLastSnapshotNumber(
                 );
                 await delay(rateLimitReset * 1000);
               }
-              // Comment out this code if ou are developing locally
+              // Comment out this code if uou are developing locally
```
Comment (goldblatt) [inline scripts/alert_emails/send_emails.ts:193]: you**

File: scripts/alert_emails/send_emails.ts [hunk 3]
```diff
@@ -196,24 +200,33 @@ async function setLastSnapshotNumber(
...
-              console.log(err);
-              errorCount += 1;
+            .catch(async (err: CampaignMonitorError) => {
+              // if the email is invalid, move the document to the invalid collection
+              if (err.Code === CM_INVALID_EMAIL_ERROR_CODE && err.Message === CM_INVALID_EMAIL_MESSAGE) {
+                invalidEmailCount += 1;
+                const currentData = (await db.collection("alerts-subscriptions").doc(doc.id).get()).data();
+                if (currentData) {
+                  await db.collection("invalid-alerts-subscriptions").doc(doc.id).set(currentData)
+                  await db.collection("alerts-subscriptions").doc(doc.id).delete();
```
Comment (mikelehen) [inline scripts/alert_emails/send_emails.ts:210]: Would you be up for using a WriteBatch so this is atomic? 😁  Something like (untested):

```
await db.batch()
    .set(db.collection("invalid-alerts-subscriptions").doc(doc.id), currentData)
    .delete(db.collection("alerts-subscriptions").doc(doc.id))
    .commit();
```

Comment (mikelehen) [reply scripts/alert_emails/send_emails.ts:210]: though I guess it really doesn't matter much...  it's just a great use case for write batches. 😁

**Pull request merged**